### DISCUSSION TITLE
Get metrics (errors/warnings/etc. in the last minute) using API

### DISCUSSION THREAD

**Discussion by DNF-SaS** [datalust/seq-tickets#1489] [Help]
Hi,

I'm using SEQ for receiving logmessages from various software-systems in a centralized manner.
Monitoring of Servers and other system components is done via PRTG.
Now I'd like to include the number of certain events received by SEQ in PRTG.
It is possible to make PRTG start a timer-based API-call every minute and parse the result (if provided as JSON) into it's monitoring-channels.

**Question**: can I use SEQ-API to query the described metric (e.g. received errors in the last minute) and get the result as JSON?

Thanks for any help!

Regards
Sascha

**Comment by nblumhardt** [comment]
Hi Sascha - thanks for your message.

The `/api/data` endpoint usually covers these kinds of cases; the routes it accepts look like:

```
api/data{?q,signal,rangeStartUtc,rangeEndUtc,defer,format,timeoutMS}
```

(Note, it's important to provide that timeout :-))

E.g., assuming you've created a signal for "Errors", a GET request like:

```
https://seq.example.com/api/data?q=select%20count(*)%20from%20stream&signal=signal-1540&rangeStartUtc=2022-02-17T14%3A00%3A00.000Z&rangeEndUtc=2022-02-17T15%3A00%3A00.000Z&timeoutMS=120000
```

Will return results like:

```json
{"Columns":["(count)"],"Rows":[[17]],"Statistics":{...}}
```

You'll need to add either an `apiKey=` parameter to the URL, or an `X-Seq-ApiKey` to the GET request, to authenticate.

Let me know if this helps!

Best regards,
Nick

**Comment by DNF-SaS** [comment]
Hi Nick,

thanks for your fast reply.
~~Unfortunately, our monitoring system is not able to provide values for rangeStartUtc and rangeEndUtc.
Is it possible to tell SEQ to simply use the last minute (e.g. the moment of API-call minus 60 seconds)?~~
We managed adding the range by wrapping the API-call into a custom plugin returning SEQ's API-Data.
Can we get the count of more than one signal in a single call using the API?

Regards
Sascha

**Reply by nblumhardt** [reply]
Hi Sascha,

That's good to hear!

It isn't possible currently to query more than one signal in a single API call.

Best regards,
Nick
